Framework/Reflectometry/src/ReflectometryReductionOneAuto3.cpp

```diff
@@ -948,22 +963,80 @@ auto ReflectometryReductionOneAuto3::getOutputNamesForGroupMember(const std::vec
   return outputNames;
 }
 
+/** Find the polarization correction method to use for the given efficiencies workspace.
+ * Checks the workspace axes labels to determine the appropriate correction method.
+ * We don't check all the workspace labels here, we just look for the first label that matches
+ * one of the efficiency factors for a supported correction method.
+ */
+std::string
+ReflectometryReductionOneAuto3::findPolarizationCorrectionMethod(const API::MatrixWorkspace_sptr &efficiencies) {
+  try {
+    auto const &axis = dynamic_cast<TextAxis &>(*efficiencies->getAxis(1));
+
+    for (size_t i = 0; i < axis.length(); ++i) {
+      if (std::find(CorrectionMethod::WILDES_AXES.begin(), CorrectionMethod::WILDES_AXES.end(), axis.label(i)) !=
+          CorrectionMethod::WILDES_AXES.end()) {
+        return CorrectionMethod::WILDES;
+      }
+      if (std::find(CorrectionMethod::FREDRIKZE_AXES.begin(), CorrectionMethod::FREDRIKZE_AXES.end(), axis.label(i)) !=
+          CorrectionMethod::FREDRIKZE_AXES.end()) {
+        return CorrectionMethod::FREDRIKZE;
+      }
+    }
+  } catch (std::bad_cast &) {
+    throw std::runtime_error("Efficiencies workspace is not in a supported format");
+  }
+  throw std::runtime_error(
+      "Axes labels for efficiencies workspace do not match any supported polarization correction method");
+}
+
+/** Find the polarization correction option to use for the given correction method, based on the number of workspaces
+ * in the input workspace group.
+ * For the Fredrikze algorithm the correction option represents the polarization analysis mode.
+ * For Wildes the correction option is the flipper configuration. There are more than two possible flipper
+ * configurations, but this is mainly intended for use by POLREF initially so we only need to support two for now.
+ */
+std::string ReflectometryReductionOneAuto3::findPolarizationCorrectionOption(const std::string &correctionMethod) {
+  auto groupIvsLam =
+      AnalysisDataService::Instance().retrieveWS<WorkspaceGroup>(getPropertyValue("OutputWorkspaceWavelength"));
+  auto numWorkspacesInGrp = groupIvsLam->size();
+  if (numWorkspacesInGrp == 2) {
+    return (correctionMethod == CorrectionMethod::FREDRIKZE) ? CorrectionOption::PNR
+                                                             : CorrectionOption::FLIPPERS_NO_ANALYSER;
+  } else if (numWorkspacesInGrp == 4) {
```

Comment:
I don't think this needs to be `else if` if the previous statement always returns

Reply:
I sorted this in the end while I was moving parameters around 🙂